src/openforms/submissions/models/submission_step.py

```diff
@@ -196,7 +196,15 @@ def can_submit(self) -> bool:
 
     @property
     def is_applicable(self) -> bool:
-        return self._is_applicable
+        if self._is_applicable is not None:
+            return self._is_applicable
+        return self.form_step.is_applicable
+
+    @is_applicable.setter
+    def is_applicable(self, value: bool) -> None:
+        if not isinstance(value, bool):
+            raise ValueError(f"'is_applicable' expects a boolean value, got {value}")
```

Comment:
In which case is this not a bool?

Reply:
Currently none (hopefully), but it might be better to have this `ValueError` raised here with a nice message instead of an unhandled exception somewhere else?

Reply:
it's still an unhandled exception, no? The zen of python states "it's better to ask for forgiveness than permission". I prefer not having this sort of defensive "what if" programming unless there's a good concrete reason.

Reply:
> Additionally - this is really the domain of the type checker to prevent mistakes/bugs like that.

True but in this case `is_applicable` could be set dynamically when applying some kind of logic. I'll make the change